crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -463,62 +797,66 @@ async fn register_agent_activity(
     check_prev_action(action)?;
     check_valid_if_dna(action, dna_def)?;
     if let Some(prev_action_hash) = prev_action_hash {
-        check_and_hold_register_agent_activity(
-            prev_action_hash,
-            cascade,
-            incoming_dht_ops_sender,
-            |_| Ok(()),
-        )
-        .await?;
+        // Just make sure we have the dependency and if not then don't mark this action as valid yet
+        let mut validation_dependencies = validation_dependencies.lock();
```

Comment:
Access can now just be done immediately without waiting on network fetches. If it's missing it'll be reported as such and the workflow is responsible for going out to the network and fetching it later